File: src/cs/Samples/Samples/Samples/SolidEdge/Draft/SaveWindowAsImage.cs
using SolidEdgeCommunity; //SolidEdge.Community.dll
using SolidEdgeFramework.Extensions; //SolidEdge.Community.dll
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ApiSamples.Samples.SolidEdge.Draft
{
    /// <summary>
    /// Saves the active 2D window as an image.
    /// </summary>
    class SaveWindowAsImage
    {
        static void RunSample(bool breakOnStart)
        {
            if (breakOnStart) System.Diagnostics.Debugger.Break();

            SolidEdgeFramework.Application application = null;
            SolidEdgeFramework.Documents documents = null;
            SolidEdgeFramework.SolidEdgeDocument document = null;
            SolidEdgeDraft.SheetWindow sheetWindow = null;

            string[] extensions = { ".jpg", ".bmp", ".tif" };
            Guid guid = Guid.NewGuid();
            string folder = Environment.GetFolderPath(Environment.SpecialFolder.DesktopDirectory);

            try
            {
                // Register with OLE to handle concurrency issues on the current thread.
                OleMessageFilter.Register();

                // Connect to or start Solid Edge.
                application = SolidEdgeCommunity.SolidEdgeInstall.Connect(true);

                // Make sure user can see the GUI.
                application.Visible = true;

                // Bring Solid Edge to the foreground.
                application.Activate();

                // Get a reference to the Documents collection.
                documents = application.Documents;

                // Get a reference to the active document.
                document = application.GetActiveDocument<SolidEdgeFramework.SolidEdgeDocument>(false);

                // Make sure we have a document.
                if (document == null)
                {
                    throw new System.Exception(Resources.NoActiveDocument);
                }

                // 2D windows are of type SolidEdgeDraft.SheetWindow.
                sheetWindow = application.ActiveWindow as SolidEdgeDraft.SheetWindow;

                if (sheetWindow != null)
                {
                    // Save each extension.
                    foreach (string extension in extensions)
                    {
                        // File saved to desktop.
                        string filename = System.IO.Path.ChangeExtension(guid.ToString(), extension);
                        filename = System.IO.Path.Combine(folder, filename);

                        double resolution = 1;  // DPI - Larger values have better quality but also lead to larger file.
                        int colorDepth = 24;
                        int width = sheetWindow.UsableWidth;
                        int height = sheetWindow.UsableHeight;

                        // You can specify .bmp (Windows Bitmap), .tif (TIFF), or .jpg (JPEG).
                        sheetWindow.SaveAsImage(
                            FileName: filename,
                            Width: width,
                            Height: height,
                            Resolution: resolution,
                            ColorDepth: colorDepth,
                            ImageQuality: SolidEdgeFramework.SeImageQualityType.seImageQualityHigh,
                            Invert: false);

                        Console.WriteLine("Saved '{0}'.", filename);
                    }
                }
                else
                {
                    throw new System.Exception(Resources.NoActive2dWindow);
                }
            }
            catch (System.Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
            finally
            {
                OleMessageFilter.Unregister();
            }
        }
    }
}

File: vb/SolidEdge/ApiDemos/ApiDemos/Samples/Draft/SaveWindowAsImage.vb
Imports SolidEdgeCommunity.Extensions ' Enabled extension methods from SolidEdge.Community.dll
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace Draft
	''' <summary>
	''' Saves the active 2D window as an image.
	''' </summary>
	Friend Class SaveWindowAsImage
		Private Shared Sub RunSample(ByVal breakOnStart As Boolean)
			If breakOnStart Then
				System.Diagnostics.Debugger.Break()
			End If

			Dim application As SolidEdgeFramework.Application = Nothing
			Dim documents As SolidEdgeFramework.Documents = Nothing
			Dim document As SolidEdgeFramework.SolidEdgeDocument = Nothing
			Dim sheetWindow As SolidEdgeDraft.SheetWindow = Nothing

			Dim extensions() As String = { ".jpg", ".bmp", ".tif" }
			Dim guid As Guid = System.Guid.NewGuid()
			Dim folder As String = Environment.GetFolderPath(Environment.SpecialFolder.DesktopDirectory)

			Try
				' Register with OLE to handle concurrency issues on the current thread.
				SolidEdgeCommunity.OleMessageFilter.Register()

				' Connect to or start Solid Edge.
				application = SolidEdgeCommunity.SolidEdgeUtils.Connect(True, True)

				' Bring Solid Edge to the foreground.
				application.Activate()

				' Get a reference to the Documents collection.
				documents = application.Documents

				' Get a reference to the active document.
				document = application.GetActiveDocument(Of SolidEdgeFramework.SolidEdgeDocument)(False)

				' Make sure we have a document.
				If document Is Nothing Then
					Throw New System.Exception(Resources.NoActiveDocument)
				End If

				' 2D windows are of type SolidEdgeDraft.SheetWindow.
				sheetWindow = TryCast(application.ActiveWindow, SolidEdgeDraft.SheetWindow)

				If sheetWindow IsNot Nothing Then
					' Save each extension.
					For Each extension As String In extensions
						' File saved to desktop.
						Dim filename As String = System.IO.Path.ChangeExtension(guid.ToString(), extension)
						filename = System.IO.Path.Combine(folder, filename)

						Dim resolution As Double = 1 ' DPI - Larger values have better quality but also lead to larger file.
						Dim colorDepth As Integer = 24
						Dim width As Integer = sheetWindow.UsableWidth
						Dim height As Integer = sheetWindow.UsableHeight

						' You can specify .bmp (Windows Bitmap), .tif (TIFF), or .jpg (JPEG).
						sheetWindow.SaveAsImage(FileName:= filename, Width:= width, Height:= height, Resolution:= resolution, ColorDepth:= colorDepth, ImageQuality:= SolidEdgeFramework.SeImageQualityType.seImageQualityHigh, Invert:= False)

						Console.WriteLine("Saved '{0}'.", filename)
					Next extension
				Else
					Throw New System.Exception(Resources.NoActive2dWindow)
				End If
			Catch ex As System.Exception
				Console.WriteLine(ex.Message)
			Finally
				SolidEdgeCommunity.OleMessageFilter.Unregister()
			End Try
		End Sub
	End Class
End Namespace
